pkg/segment/reader/segread/segstatsreader.go

```diff
@@ -143,37 +143,52 @@ func readSingleSst(fdata []byte, qid uint64) (*structs.SegStats, error) {
 
 	idx += hllSize
 
+	if version == utils.VERSION_SEGSTATS_BUF_V4[0] {
+		err := readCurrentSST(&sst, fdata, idx)
+		if err != nil {
+			return nil, fmt.Errorf("readSingleSst: error reading current sst, err: %v", err)
+		}
+		return &sst, nil
+	}
+
 	if !sst.IsNumeric {
 		return &sst, nil
 	}
 
+	readNumericStats(&sst, fdata, idx, false)
+
+	return &sst, nil
+}
+
+func readNumericStats(sst *structs.SegStats, fdata []byte, idx uint32, readNumericCount bool) {
 	sst.NumStats = &structs.NumericStats{}
-	// read Min Ntype
-	min := utils.NumTypeEnclosure{}
-	min.Ntype = utils.SS_DTYPE(fdata[idx : idx+1][0])
+
+	min := utils.CValueEnclosure{}
+	// read Min Dtype
+	min.Dtype = utils.SS_DTYPE(fdata[idx : idx+1][0])
 	idx += 1
-	if min.Ntype == utils.SS_DT_FLOAT {
-		min.FloatVal = toputils.BytesToFloat64LittleEndian(fdata[idx : idx+8])
+	if min.Dtype == utils.SS_DT_FLOAT {
+		min.CVal = toputils.BytesToFloat64LittleEndian(fdata[idx : idx+8])
 	} else {
-		min.IntgrVal = toputils.BytesToInt64LittleEndian(fdata[idx : idx+8])
+		min.CVal = toputils.BytesToInt64LittleEndian(fdata[idx : idx+8])
 	}
-	sst.NumStats.Min = min
+	sst.Min = min
 	idx += 8
 
-	// read Max Ntype
-	max := utils.NumTypeEnclosure{}
-	max.Ntype = utils.SS_DTYPE(fdata[idx : idx+1][0])
+	max := utils.CValueEnclosure{}
+	// read Max Dtype
+	max.Dtype = utils.SS_DTYPE(fdata[idx : idx+1][0])
 	idx += 1
-	if max.Ntype == utils.SS_DT_FLOAT {
-		max.FloatVal = toputils.BytesToFloat64LittleEndian(fdata[idx : idx+8])
+	if max.Dtype == utils.SS_DT_FLOAT {
+		max.CVal = toputils.BytesToFloat64LittleEndian(fdata[idx : idx+8])
 	} else {
-		max.IntgrVal = toputils.BytesToInt64LittleEndian(fdata[idx : idx+8])
+		max.CVal = toputils.BytesToInt64LittleEndian(fdata[idx : idx+8])
 	}
-	sst.NumStats.Max = max
+	sst.Max = max
 	idx += 8
 
-	// read Sum Ntype
 	sum := utils.NumTypeEnclosure{}
+	// read Sum Ntype
```

Comment:
update/remove this comment or clarify why Ntype means

Reply:
`NType` is already a field in NumTypeEnclosure.